### PR TITLE
feat: server setup complete event

### PR DESCRIPTION
**Pull request by vivonk**
## Description
Send an analytics events on instance registration

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/cloud-services/issues/1232

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/InstanceConfigHelperCEImpl.java [hunk 1]
```diff
@@ -80,12 +89,41 @@ public Mono<? extends Config> registerInstance() {
...
+                            .switchIfEmpty(Mono.defer(() -> {
+                                sendServerSetupEvent(instanceId);
+                                return Mono.just(new Config());
+                            }))
+                            .flatMap(config -> {
+                                // if instance isn't already marked registered
+                                if (config.getConfig() != null
+                                        && !Boolean.parseBoolean(
+                                                (String) config.getConfig().get("value"))) {
+                                    sendServerSetupEvent(instanceId);
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/InstanceConfigHelperCEImpl.java:105]: Will this result in firing the same event couple of times if the entry is not present in config collection?

**Review by abhvsn — COMMENTED** [review]
Minor comment otherwise LGTM

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/InstanceConfigHelperCEImpl.java (hunk 1 above)
Comment (vivonk) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/InstanceConfigHelperCEImpl.java:105]: Nope, this event will be fired only once

**Review by abhvsn — APPROVED** [review]

**Pull request merged**